### PR TITLE
refactor(celery-beat): remove cron jobs, subs. by beat tasks 

### LINKED ISSUE #1: Test tweet captures and weather data captures to deploy to production
- [x] Rename filenames, directories and docker services.
- [x] Remove CRON.
- [x] Run task tests using CELERY-BEAT.

### PR DESCRIPTION
**Pull request by luabida**
fix #1 

All cron jobs were removed from `crawlclima`, the containers were refactored as well.

The module `utilities` is still left behind, but it seems to be in the wrong repo. Willing to remove the entire `utilities` directory and keep only the necessary scripts to run crawlclima.

Another issue while refactoring, as the tasks use `self` in their calls, I don't see how and why it's possible to call the tasks via CLI. But if this system will be used somehow this way, we can thing in a workaround for calling the methods ignoring the `self` attr. used by the task (I've seem only being used to retry() the task if failed, but it needs a deeper method refactoring to remove the self from the methods)

### CODE REVIEW

**Comment by luabida** [comment]
Current traceback for `pega_tweets`:

```
2022-12-27 16:48:05 [2022-12-27 19:48:05,541: WARNING/ForkPoolWorker-4] --- Logging error ---
2022-12-27 16:48:05 [2022-12-27 19:48:05,547: WARNING/ForkPoolWorker-4] Traceback (most recent call last):
2022-12-27 16:48:05 [2022-12-27 19:48:05,547: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connection.py", line 174, in _new_conn
2022-12-27 16:48:05     conn = connection.create_connection(
2022-12-27 16:48:05            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,547: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/util/connection.py", line 72, in create_connection
2022-12-27 16:48:05     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
2022-12-27 16:48:05                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,548: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/socket.py", line 961, in getaddrinfo
2022-12-27 16:48:05     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
2022-12-27 16:48:05                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,548: WARNING/ForkPoolWorker-4] socket.gaierror: [Errno -2] Name or service not known
2022-12-27 16:48:05 [2022-12-27 19:48:05,548: WARNING/ForkPoolWorker-4] 
2022-12-27 16:48:05 During handling of the above exception, another exception occurred:
2022-12-27 16:48:05 [2022-12-27 19:48:05,548: WARNING/ForkPoolWorker-4] Traceback (most recent call last):
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connectionpool.py", line 703, in urlopen
2022-12-27 16:48:05     httplib_response = self._make_request(
2022-12-27 16:48:05                        ^^^^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connectionpool.py", line 398, in _make_request
2022-12-27 16:48:05     conn.request(method, url, **httplib_request_kw)
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connection.py", line 239, in request
2022-12-27 16:48:05     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/http/client.py", line 1282, in request
2022-12-27 16:48:05     self._send_request(method, url, body, headers, encode_chunked)
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/http/client.py", line 1328, in _send_request
2022-12-27 16:48:05     self.endheaders(body, encode_chunked=encode_chunked)
2022-12-27 16:48:05 [2022-12-27 19:48:05,549: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/http/client.py", line 1277, in endheaders
2022-12-27 16:48:05     self._send_output(message_body, encode_chunked=encode_chunked)
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/http/client.py", line 1037, in _send_output
2022-12-27 16:48:05     self.send(msg)
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/http/client.py", line 975, in send
2022-12-27 16:48:05     self.connect()
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connection.py", line 205, in connect
2022-12-27 16:48:05     conn = self._new_conn()
2022-12-27 16:48:05            ^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connection.py", line 186, in _new_conn
2022-12-27 16:48:05     raise NewConnectionError(
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4] urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fa7650c2d50>: Failed to establish a new connection: [Errno -2] Name or service not known
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4] 
2022-12-27 16:48:05 During handling of the above exception, another exception occurred:
2022-12-27 16:48:05 [2022-12-27 19:48:05,550: WARNING/ForkPoolWorker-4] Traceback (most recent call last):
2022-12-27 16:48:05 [2022-12-27 19:48:05,551: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/requests/adapters.py", line 489, in send
2022-12-27 16:48:05     resp = conn.urlopen(
2022-12-27 16:48:05            ^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,551: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/connectionpool.py", line 787, in urlopen
2022-12-27 16:48:05     retries = retries.increment(
2022-12-27 16:48:05               ^^^^^^^^^^^^^^^^^^
2022-12-27 16:48:05 [2022-12-27 19:48:05,551: WARNING/ForkPoolWorker-4]   File "/opt/conda/envs/crawlclima/lib/python3.11/site-packages/urllib3/util/retry.py", line 592, in increment
2022-12-27 16:48:05     raise MaxRetryError(_pool, url, error or ResponseError(cause))
2022-12-27 16:48:05 [2022-12-27 19:48:05,551: WARNING/ForkPoolWorker-4] urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='observatorio.inweb.org.br', port=80): Max retries exceeded with url: /dengueapp/api/1.0/totais?cidade=110001&cidade=110037&cidade=110040&cidade=110034&cidade=110002&cidade=110045&cidade=110003&cidade=110060&cidade=110004&cidade=110070&cidade=110080&cidade=110090&cidade=110005&cidade=110092&cidade=110006&cidade=110007&cidade=110008&cidade=110094&cidade=110009&cidade=110100&cidade=110010&cidade=110110&cidade=110011&cidade=110012&cidade=110013&cidade=110120&cidade=110130&cidade=110140&cidade=110014&cidade=110033&cidade=110143&cidade=110050&cidade=110015&cidade=110145&cidade=110018&cidade=110146&cidade=110020&cidade=110025&cidade=110147&cidade=110026&cidade=110028&cidade=110029&cidade=110148&cidade=110149&cidade=110032&cidade=110150&cidade=110155&cidade=110160&cidade=110170&cidade=110175&inicio=2022-12-19T00:00:00&fim=2022-12-27T00:00:00&token=XXXXX (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa7650c2d50>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

**Comment by luabida** [comment]
@esloch I still have no idea what to do with the module `utilities`, please let me know if there is something in there that could be used in crawlclima.

**Comment by esloch** [comment]
> @esloch I still have no idea what to do with the module `utilities`, please let me know if there is something in there that could be used in crawlclima.

Utilities contain tools and CLI applications to prepare and execute tweets and weather data captures in the localhost environment, without the need to containerize the services.
Includes:
- Dumps the database schemas.
- Inserts population data from IBGE.
- Download and import the territorial mesh with geospatial data for all Brazilian municipalities by state into the database.
- Exports to files in JSON formats the polygons for creating maps.
- Inserts the meteorological stations (REDEMET) of Brazilian airports for each municipality.

https://github.com/AlertaDengue/AlertaDengueCaptura/blob/main/crawlclima/utilities/README.md
https://github.com/AlertaDengue/AlertaDengueCaptura/blob/main/Makefile#L82

**Comment by luabida** [comment]
Still needs tests

File: crawlclima/redmet/rmet.py [hunk 1]
```diff
@@ -9,8 +9,8 @@
 import pandas as pd
 import psycopg2
 import requests
-from crawlclima import config
-from crawlclima.utils.models import save
+from crawlclima.config import dbconnections
+from crawlclima.redmet.models import save
```
Comment (esloch) [inline crawlclima/redmet/rmet.py]: typo:
[*redemet*](https://www.redemet.aer.mil.br/)

File: tests/unit/test_tasks.py [hunk 2]
```diff
@@ -32,14 +32,14 @@ def test_task_pegatweets(self):
         self.assertEqual(res, 200)
         self.assertGreater(len(resp), 0)
 
-    def test_task_fetch_redemet(self):
```
Comment (esloch) [inline tests/unit/test_tasks.py]: **test_task_fetch_redemet()** Is a valid name because there is more than one weather variable to capture...

**Review by esloch — APPROVED** [review]
LGTM!

File: crawlclima/celeryapp.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from celery.schedules import crontab
+
+app = Celery("crawlclima")
+
+app.config_from_object("crawlclima.config.celeryconfig")
+
+
+# Celery Beat Scheduler
+app.conf.beat_schedule = {
+    "captura-temperatura-redmet": {
```
Comment (esloch) [inline crawlclima/celeryapp.py]: typo: "captura-temperatura-redemet"...

**Comment by esloch** [comment]
Great job @luabida!
Maybe disable the tweets schedule for the moment, as we don't have a response from the UFMG observatory.
What do you think @fccoelho?

**Comment by fccoelho** [comment]
Yes the tweets are not very likely to work for a while

**Comment by luabida** [comment]
Ok, thank you @esloch and @fccoelho, I've commented out the tweets task and fixed the typo.

**Comment by esloch** [comment]
@luabida, Here is our [*logs*](https://github.com/osl-incubator/CrawlClima/blob/131c01df183df3e11d62d768618ccb8824bdc24b/containers/Dockerfile#L22) friend...

**Comment by esloch** [comment]
LGTM!

**Pull request merged**